lib/ai/tools/utils/pty-output.ts

```
@@ -0,0 +1,89 @@
+ * metadata (machineid, hostname, bootid, pid, cwd, etc.). When these are
+ * split across PTY data chunks the payload text leaks into the visible
+ * output. These helpers remove that noise while **preserving ANSI escape
+ * sequences** (colors, cursor, etc.) for Shiki ANSI rendering in the UI.
+ */
+
+// Pre-compiled regexes — avoids re-creation on every call.
+// Each pattern requires the ESC (\x1b) prefix so we never accidentally match
+// literal text that happens to contain `]3008;` or `[?2004`.
+
+/** VS Code shell-integration: \x1b]633;…BEL or \x1b]633;…ST */
+const OSC_633_RE = /\x1b\]633;[^\x07\x1b]*(?:\x07|\x1b\\)\r?\n?/g;
```

Biome (2.3.13)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)